wdl-lint/src/rules/malformed_lint_directive.rs

```diff
@@ -0,0 +1,137 @@
+    let accepted_directives = ACCEPTED_LINT_DIRECTIVES.join(", ");
+    Diagnostic::warning("lint directive not recognized")
+        .with_rule(ID)
+        .with_label("lint directive not recognized", span)
+        .with_fix(format!(
+            "use any of the recognized lint directives: [{:#?}]",
+            accepted_directives
+        ))
+}
+
+/// Creates a "Missing Whitespace" diagnostic.
+fn missing_whitespace(span: Span) -> Diagnostic {
+    Diagnostic::warning("expected exactly one space before lint directive")
+        .with_rule(ID)
+        .with_label("missing whitespace", span)
+}
+
+/// Creates a "No Colon Detected" diagnostic.
+fn no_colon_detected(span: Span) -> Diagnostic {
+    Diagnostic::warning("expected colon following a lint directive")
+        .with_rule(ID)
+        .with_label("no colon detected", span)
+}
+
+/// Detects a malformed lint directive.
+#[derive(Default, Debug, Clone, Copy)]
+pub struct MalformedLintDirectiveRule;
+
+impl Rule for MalformedLintDirectiveRule {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Flags malformed lint directives."
+    }
+
+    fn explanation(&self) -> &'static str {
+        "Comments which begin with `#@` must only contain valid lint directives. Lint directives \
+         must be on their own line, only preceded by whitespace. Lint directives should follow the \
+         pattern `#@ <directive>: <value>` _exactly_. Currently the only accepted lint directive \
+         is `except`. For example, `#@ except: MalformedLintDirective`."
+    }
+
+    fn tags(&self) -> TagSet {
+        TagSet::new(&[Tag::Clarity, Tag::Correctness])
+    }
+
+    fn exceptable_nodes(&self) -> Option<&'static [wdl_ast::SyntaxKind]> {
+        None
+    }
+}
+
+impl Visitor for MalformedLintDirectiveRule {
+    type State = Diagnostics;
+
+    fn document(&mut self, _: &mut Self::State, _: VisitReason, _: &Document, _: SupportedVersion) {
+        // This is intentionally empty, as this rule has no state.
+    }
+
+    fn comment(&mut self, state: &mut Self::State, comment: &Comment) {
+        if let Some(lint_directive) = comment.as_str().strip_prefix("#@") {
+            let base_offset = comment.span().start();
+
+            if is_inline_comment(comment) {
+                state.add(inline_lint_directive(comment.span()));
+            }
+
+            if !lint_directive.starts_with(" ") {
+                state.add(missing_whitespace(Span::new(base_offset + 2, 1)));
+            }
+
+            if lint_directive.starts_with("  ") {
+                let leading_whitespace_len =
+                    lint_directive.len() - lint_directive.trim_start().len();
+                state.add(excessive_whitespace(Span::new(
+                    base_offset + 2,
+                    leading_whitespace_len,
+                )));
+            }
+
+            let mut directive = lint_directive.trim().split(" ").next().expect("directive");
```

Comment:
Excellent, I've added that new diagnostic earlier in the rule logic (before whitespace checking), but am curious: should I just `unwrap()` the value here instead? is there a better way to unwrap the Option returned by the iterator that avoids panicking? or is the thinking that with the check earlier in the function, this unwrap will never panic?

Reply:
So there's a bunch of ways to handle the `None` variant of `Option` without using `unwrap()/expect()`.
Some examples:
```
match <>.next() {
    Some(val) => <use the returned val>,
    None => <handle the None case>,
}
```
or
```
if let Some(val) = <>.next() {
    <use val>
}
<we are ignoring the None case and doing nothing with it>
```
or
```
let val = <>.next().unwrap_or(<default val>);
```
And there's more too!

------------
@peterhuene I just checked out the `split()` method, and if I'm reading it correctly I think Michael's `expect()` here is actually fine. `split()` will always return at least one element, so calling `next()` on it should always yield `Some()` (although the returned str might be empty, which I think might cause a panic a few lines later).

I think it should get a rewrite though just because that's not very clear. @michaelgattas can you take another stab at writing this without using `expect()` or `unwrap()`?

Reply:
@a-frantz you're right, for some reason I thought `split` on an empty string would return an empty iterator. It won't panic here then.